l5-operator/controllers/bestie_controller.go

```diff
@@ -73,28 +74,82 @@ func (r *BestieReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
-	var result *reconcile.Result
+	//var result *reconcile.Result
 
-	result, err = r.ensureJob(ctx, bestie, r.bestieJob(bestie))
-	if result != nil {
-		return *result, err
+	// reconcile Postgres
+	pgo := &pgov1.PostgresCluster{}
+
+	err = r.Get(ctx, types.NamespacedName{Name: bestie.Name + "-pgo", Namespace: bestie.Namespace}, pgo)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating a new PGC for bestie")
+			fileName := "config/resources/postgrescluster.yaml"
+			r.applyManifests(ctx, req, bestie, pgo, fileName)
+		} else {
+			return ctrl.Result{Requeue: true}, err
+		}
+		// TODO: should we update then?
 	}
 
-	result, err = r.ensureDeployment(ctx, bestie, r.bestieAppDeployment(bestie))
-	if result != nil {
-		return *result, err
+	// reconcile Deployment
+	dp := &appsv1.Deployment{}
+
+	err = r.Get(ctx, types.NamespacedName{Name: bestie.Name + "-app", Namespace: bestie.Namespace}, dp)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating a new app for bestie")
+			fileName := "config/resources/bestie-deploy.yaml"
+			r.applyManifests(ctx, req, bestie, dp, fileName)
+		} else {
+			return ctrl.Result{Requeue: true}, err
+		}
+		// TODO: should we update then?
 	}
 
-	//create a job to migrate the dabase
+	job := &batchv1.Job{}
 
-	result, err = r.ensureService(ctx, bestie, r.bestieAppService(bestie))
-	if result != nil {
-		return *result, err
+	err = r.Get(ctx, types.NamespacedName{Name: bestie.Name + "-job", Namespace: bestie.Namespace}, job)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating a new job for bestie")
+			fileName := "config/resources/bestie-job.yaml"
+			r.applyManifests(ctx, req, bestie, job, fileName)
+		} else {
+			return ctrl.Result{Requeue: true}, err
+		}
+		// TODO: should we update then?
 	}
 
-	result, err = r.ensureRoute(ctx, bestie, r.bestieRoute(bestie))
-	if result != nil {
-		return *result, err
+	// reconcile service
+
+	svc := &corev1.Service{}
+
+	err = r.Get(ctx, types.NamespacedName{Name: bestie.Name + "-service", Namespace: bestie.Namespace}, svc)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating a new service for bestie")
+			fileName := "config/resources/bestie-svc.yaml"
+			r.applyManifests(ctx, req, bestie, svc, fileName)
+		} else {
+			return ctrl.Result{Requeue: true}, err
+		}
+		// TODO: should we update then?
+	}
+
+	// reconcile route
+
+	route := &routev1.Route{}
+
+	err = r.Get(ctx, types.NamespacedName{Name: bestie.Name + "-route", Namespace: bestie.Namespace}, route)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("Creating a new route for bestie")
+			fileName := "config/resources/bestie-route.yaml"
+			r.applyManifests(ctx, req, bestie, route, fileName)
```

Comment:
since we are just applying manifests is there anything stopping us from just applying all the manifests at a path instead of individually referencing them and applying them one by one ?